File: Dev/unitTest_Engine_cs/Graphics/2D/TextObject2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics._2D
{
    class TextObject2D : EngineTest
    {
        public TextObject2D()
            : base(20)
        {
        }

        protected override void OnStart()
        {
            var scene = new asd.Scene();
            var layer = new asd.Layer2D();
            var obj = new asd.TextObject2D();

            Assert.AreNotEqual<asd.Scene>(scene, null);
            Assert.AreNotEqual<asd.Layer>(layer, null);
            Assert.AreNotEqual<asd.TextObject2D>(obj, null);

            asd.Engine.ChangeScene(scene);
            scene.AddLayer(layer);
            layer.AddObject(obj);

			var font = asd.Engine.Graphics.CreateFont("Data/Font/Nac0812.aff");

            obj.Font = font;
			obj.Text = "DependencyPropertyは\n依存関係プロパティのこと";

            //object->SetCenterPosition(Vector2DF(128, 128));
            obj.Position = new asd.Vector2DF(0, 0);
        }

        protected override void OnUpdating()
        {

        }
    }
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/2D/TextObject2D.cpp
#include "../../EngineTest.h"
#include <Altseed.h>
#include <gtest/gtest.h>
#include <memory>

using namespace std;
using namespace asd;

class Graphics_TextObject2D : public EngineTest
{
public:
	Graphics_TextObject2D(bool isOpenGLMode) : EngineTest(u"TextObject2D", isOpenGLMode, 10) {}

protected:
	void OnStart()
	{
		auto scene = make_shared<Scene>();
		auto layer = make_shared<Layer2D>();
		auto object = make_shared<TextObject2D>();

		ASSERT_NE(scene, nullptr);
		ASSERT_NE(layer, nullptr);
		ASSERT_NE(object, nullptr);

		asd::Engine::ChangeScene(scene);
		scene->AddLayer(layer);

		auto g = asd::Engine::GetGraphics();

		asd::Engine::GetFile()->AddRootDirectory(u"Data");

		layer->AddObject(object);
		auto font = g->CreateFont(u"Font/Nac0812.aff");

		object->SetText(u"DependencyPropertyとは");

		// object->SetCenterPosition(Vector2DF(128, 128));
		object->SetPosition(Vector2DF(0, 0));
		object->SetFont(font);

		auto size = font->CalcTextureSize(u"DependencyPropertyとは", WritingDirection::Horizontal);

		printf("%d %d\n", size.X, size.Y);
	}
};

ENGINE_TEST(Graphics, TextObject2D)
